### PR TITLE
Android Completion iOS update

### PR DESCRIPTION
**Pull request by diegogarciar**
In this pull the following improvements have been done

Example
- Example with real use case

Both
- Added client identifier for background incoming calls
- Compatibility with flutter firebase messaging (Twilio service overrides firebase service so I had to merge them)

iOS
- Updated iOS Twilio-Voice to 6.0.0

Android:
- Toggle Speaker 
- AnswerActivity: to answer the call when app is locked.
  - Answer and Reject in full screen
- BackgroundCallActivity to manage the call when screen is locked
  - Call identifier
  - mute
  - toggle speaker
- Added Sound and Vibration, used based on user prefs
- Proper handle of Microphone permissions on Android
    - While performing a call
    - While answering a call
- Check and request Xiaomi background permissions, (per device extra permissions can also be added in the future)

### CODE REVIEW

**Review by markathomas — APPROVED** [review]

**Review by ankad — APPROVED** [review]

**Comment by diegogarciar** [comment]
I had to change the android example package name to use firebase, but you can revert it to yours

**Comment by markathomas** [comment]
> I had to change the android example package name to use firebase, but you can revert it to yours

what do you mean?  why was the change necessary?  also can you please resolve the conflicts so I can merge?

**Comment by diegogarciar** [comment]
I have resolved them, but let me revert the package name

File: example/.flutter-plugins-dependencies [hunk 1]
```diff
@@ -0,0 +1 @@
....18.0+1/","dependencies":["firebase_core"]},{"name":"firebase_core","path":"/Users/diego/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_core-0.5.0/","dependencies":[]}],"linux":[],"windows":[],"web":[{"name":"cloud_functions_web","path":"/Users/diego/flutter/.pub-cache/hosted/pub.dartlang.org/cloud_functions_web-2.0.0/","dependencies":[]},{"name":"firebase_auth_web","path":"/Users/diego/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth_web-0.3.0+1/","dependencies":[]},{"name":"firebase_core_web","path":"/Users/diego/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_core_web-0.2.0/","dependencies":[]}]},"dependencyGraph":[{"name":"flutter_twilio_voice","dependencies":[]},{"name":"cloud_functions","dependencies":["firebase_core","cloud_functions_web"]},{"name":"cloud_functions_web","dependencies":[]},{"name":"firebase_auth","dependencies":["firebase_core","firebase_auth_web"]},{"name":"firebase_auth_web","dependencies":["firebase_core"]},{"name":"firebase_core","dependencies":["firebase_core_web"]},{"name":"firebase_core_web","dependencies":[]},{"name":"firebase_messaging","dependencies":["firebase_core"]}],"date_created":"2020-10-10 23:40:17.278675","version":"1.22.0"}
```
Comment (ankad) [inline example/.flutter-plugins-dependencies]: I guess, this file and some others should be added to gitignore

Comment (diegogarciar) [reply example/.flutter-plugins-dependencies]: Is actually in the example's gitignore, I don't know why its being tracked.

**Comment by diegogarciar** [comment]
I have reverted the package name, it thought it was necessary so I could test with firebase. But maybe it isn't, I'll try not overriding it next time and see what happens.

**Comment by markathomas** [comment]
Thanks @diegogarciar !

**Pull request merged**